File: leetcode/editor/en/[1442]Count Triplets That Can Form Two Arrays of Equal XOR.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int countTriplets(int[] arr) {
        int n = arr.length;
        int[] prefix = new int[n + 1];
        int count = 0;

        for (int i = 0; i < n; i++) {
            prefix[i + 1] = prefix[i] ^ arr[i];
        }

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (prefix[i] == prefix[j + 1]) {
                    count += (j - i);
                }
            }
        }

        return count;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
/*
class Solution {
    public int countTriplets(int[] arr) {
        int n = arr.length;
        int[] prefix = new int[n + 1];
        int count = 0;

        for (int i = 0; i < n; i++) {
            prefix[i + 1] = prefix[i] ^ arr[i];
        }

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (prefix[i] == prefix[j + 1]) {
                    count += (j - i);
                }
            }
        }

        return count;
    }
}
 */

File: leetcode/editor/en/[1442]Count Triplets That Can Form Two Arrays of Equal XOR.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} arr
 * @return {number}
 */
var countTriplets = function (arr) {
    const n = arr.length;
    let prefix = [arr[0]];
    let count = 0;

    for (let i = 0; i < n; i++) {
        prefix.push(prefix[i] ^ arr[i]);
    }

    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            if (prefix[i] === prefix[j + 1]) {
                count += (j - i);
            }
        }
    }

    return count;
};
//leetcode submit region end(Prohibit modification and deletion)
